File: Jan/Week 3/Extra/problem_3.py
# Day 16 Problem No 36

# 283. Move Zeroes => https://leetcode.com/problems/move-zeroes/description/?envType=study-plan-v2&envId=leetcode-75

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        idx = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[idx] = nums[i]
                idx+=1
                
        for i in range(idx, len(nums)):
            nums[i] = 0

File: Jan/Week 4/LeetCode/Problem_3.cpp
// Day 23 Problem No 51

// 11. Container With Most Water => https://leetcode.com/problems/container-with-most-water/description/?envType=study-plan-v2&envId=leetcode-75

class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0;
        int r = height.size() -1;
        int maxx = 0;
        while(l < r){
            int h = min(height[l],height[r]);
            int w = r - l;
            maxx = max(maxx, h * w);

            if(height[l] < height[r]){
                l++;
            }else{
                r--;
            }
        }
        return maxx;
    }
};